lxd/container_lxc.go

```diff
@@ -2713,16 +2682,15 @@ func (c *containerLXC) Delete() error {
 
 	shared.LogInfo("Deleting container", ctxMap)
 
-	// Initialize storage interface for the container.
-	err := c.initStorageInterface()
-	if err != nil {
-		return err
-	}
+	// Attempt to initialize storage interface for the container.
+	c.initStorageInterface()
 
 	if c.IsSnapshot() {
 		// Remove the snapshot
-		if err := c.storage.ContainerSnapshotDelete(c); err != nil {
-			shared.LogWarn("Failed to delete snapshot", log.Ctx{"name": c.Name(), "err": err})
+		if c.storage != nil {
+			if err := c.storage.ContainerSnapshotDelete(c); err != nil {
```

Comment:
For some reason we seem to have always considered snapshot delete errors to be non-fatal in there. I can make them fatal and we'll see if there was a reason for that :)